From bff2c99b1daca0d0c071781f0e83fe5c7f26eec2 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Mon, 30 Jan 2023 10:25:22 +0100
Subject: [PATCH 0172/1695] drivers: clk: export clk_get_parent_idx()

Sometimes we need to know the index of the parent clock.
This is the case when we need to save/restore the context of the clock
in power saving mode.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 35c9813bf862240c65289c12df2c05974eb4e8a8 drivers: clk: export clk_get_parent_idx()]
Change-Id: I64abc15dad6f483902c39b4bb7961f251c12ef34
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/drivers/clk/clk.c     |  4 ++--
 core/include/drivers/clk.h | 10 ++++++++++
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/core/drivers/clk/clk.c b/core/drivers/clk/clk.c
index 1212d5693..a531f7cae 100644
--- a/core/drivers/clk/clk.c
+++ b/core/drivers/clk/clk.c
@@ -217,8 +217,8 @@ struct clk *clk_get_parent(struct clk *clk)
 	return clk->parent;
 }
 
-static TEE_Result clk_get_parent_idx(struct clk *clk, struct clk *parent,
-				     size_t *pidx)
+TEE_Result clk_get_parent_idx(struct clk *clk, struct clk *parent,
+			      size_t *pidx)
 {
 	size_t i = 0;
 
diff --git a/core/include/drivers/clk.h b/core/include/drivers/clk.h
index bf62a9cb5..54813de6e 100644
--- a/core/include/drivers/clk.h
+++ b/core/include/drivers/clk.h
@@ -227,6 +227,16 @@ static inline size_t clk_get_num_parents(struct clk *clk)
  */
 struct clk *clk_get_parent_by_index(struct clk *clk, size_t pidx);
 
+/**
+ * clk_get_parent_idx - Get index of clock parent
+ *
+ * @clk: Clock which contains parent clocks
+ * @parent: Parent clock for which you want to known the index
+ * Return a TEE_Result compliant value and set index of the parent clock
+ */
+TEE_Result clk_get_parent_idx(struct clk *clk, struct clk *parent,
+			      size_t *pidx);
+
 /**
  * clk_set_parent - Set the current clock parent
  *
-- 
2.47.2

